action-server/src/listeners/dbListeners.ts

```diff
@@ -0,0 +1,163 @@
+import { readFile } from "node:fs/promises";
+import type http from "node:http";
+import * as path from "node:path";
+import type { Pool, PoolClient } from "pg";
+import { configuration } from "../config";
+import { ActionsDbManager } from "../db";
+import { ActionWorkerPool } from "../threads/workerPool";
+import type { ActionDefinitionInsertedPayload, ActionResponse, ActionRunInsertedPayload } from "../type/types";
+import { extractSchemas } from "../utils/codeRunner";
+
+let listenClient: PoolClient | undefined;
+
+async function readFileFromStore(fileName: string): Promise<string> {
+  // read file from aerie file store and return [resolve] it as a string
+  const fileStoreBasePath = configuration().ACTION_LOCAL_STORE;
+  const filePath = path.join(fileStoreBasePath, fileName);
+  console.log(`path is ${filePath}`);
+  return await readFile(filePath, "utf-8");
+}
+
+async function handleActionDefinition(payload: ActionDefinitionInsertedPayload) {
```

Comment:
Nonblocking: I suggest renaming this to match the pattern of similar endpoints in Aerie: 

```suggestion
async function refreshActionDefinitionSchema(payload: ActionDefinitionInsertedPayload) {
```

Reply:
Thank you for the suggestion. I renamed it